src/librustdoc/clean/inline.rs

```diff
@@ -447,7 +447,7 @@ crate fn build_impl(
     };
     let polarity = tcx.impl_polarity(did);
     let trait_ = associated_trait.clean(cx);
-    if trait_.def_id() == tcx.lang_items().deref_trait() {
+    if trait_.as_ref().map(|t| t.def_id()) == tcx.lang_items().deref_trait() {
```

Comment:
This doesn't seem like a very helpful change? I would rather keep GetDefId than add more churn like this.

Reply:
`GetDefId::def_id` returns an `Option`, though, but we can always get the `DefId` of a `Path`. It adds unnecessary confusion about what information is available. I also find it very confusing that `GetDefId` is implement for `Option<T: GetDefId>`. This code is much clearer IMO.